### PR TITLE
#175 support network scaling

### LINKED ISSUE #175: Support network scaling policy
Currently we only support CPU.

### PR DESCRIPTION
**Pull request by prayerslayer**
Fixes #175

This PR introduces support for network scaling policies.

### CODE REVIEW

**Comment by prayerslayer** [comment]
I started a stack with NetworkIn policy and it went up successfully.

File: senza/components/auto_scaling_group.py [hunk 1]
```diff
@@ -172,9 +172,16 @@ def component_auto_scaling_group(definition, configuration, args, info, force, a
         }
 
         metric_type = as_conf["MetricType"]
-        metricfn = globals().get('metric_{}'.format(metric_type.lower()))
-        if not metricfn:
+        # ADVICE_NEEDED: to my understanding this change breaks existing
+        # senzafiles with MetricType: cpu (lowercase) in it.
+        if metric_type not in ["CPU", "NetworkIn", "NetworkOut"]:
```
Comment (hjacobs) [inline senza/components/auto_scaling_group.py]: please re-use the keys in the dict (to be DRY)

File: senza/components/auto_scaling_group.py [hunk 2]
```diff
@@ -172,9 +172,16 @@ def component_auto_scaling_group(definition, configuration, args, info, force, a
         }
 
         metric_type = as_conf["MetricType"]
-        metricfn = globals().get('metric_{}'.format(metric_type.lower()))
-        if not metricfn:
+        # ADVICE_NEEDED: to my understanding this change breaks existing
+        # senzafiles with MetricType: cpu (lowercase) in it.
```
Comment (hjacobs) [inline senza/components/auto_scaling_group.py]: yes, please allow lowercase too to preserve compat.

File: senza/components/auto_scaling_group.py [hunk 3]
```diff
@@ -183,6 +190,111 @@ def component_auto_scaling_group(definition, configuration, args, info, force, a
...
+    }
+    try:
+        # if someone write just Threshold: 10
+        amount = int(threshold)
+        return [threshold, unit]
+    except ValueError:
+        # check if there is a space as if somebody wrote Threshold: 20 GB
+        if " " in threshold:
+            # okay, so split it
+            [amount, unit] = threshold.split(" ")
```
Comment (hjacobs) [inline senza/components/auto_scaling_group.py]: just remove the brackets on the left side and use threshold.split() (without param which will split on 1+ whitespace chars)

File: tests/test_components.py [hunk 4]
```diff
@@ -393,3 +394,81 @@ def test_component_taupage_auto_scaling_group_user_data_with_lists_and_empty_dic
...
+    assert normalize_network_threshold(None) == []
+    assert normalize_network_threshold("10") == ["10", "Bytes"]
+    assert normalize_network_threshold(10) == ["10", "Bytes"]
+    assert normalize_network_threshold("10 B") == ["10", "Bytes"]
+    assert normalize_network_threshold("10 KB") == ["10", "Kilobytes"]
+    assert normalize_network_threshold("10 MB") == ["10", "Megabytes"]
+    assert normalize_network_threshold("10 GB") == ["10", "Gigabytes"]
+    assert normalize_network_threshold("5.7 TB") == ["5.7", "Terabytes"]
+
+    try:
```
Comment (hjacobs) [inline tests/test_components.py]: you can use "with pytest.raises(...) as ex:" here (much nicer)

**Comment by prayerslayer** [comment]
`with pytest.raises` :confetti_ball: Thanks!

**Pull request merged**